Release 1.0.1 and apply vox pupuli release best practices

bastelfreak opened the pull request.

bastelfreak committed:
Release 1.0.1 and apply vox pupuli release best practices

kenyon commented on .github/workflows/release.yml:
```diff
@@ -1,33 +1,39 @@
-name: Release
+---
+name: Publish package to GitHub Packages and clojars.org
```
Seems like we shouldn't be doing clojure release steps for a ruby gem?

bastelfreak committed:
rspec-its: Allow 2.x

bastelfreak committed:
spec_helper: require rspec/collection_matchers

this was an implicit dependency and resolved until
https://github.com/puppetlabs/puppet-lint/commit/1fc39b18dcc05d44f8673bd91a609bb03c8937dc#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94L15
was merged (puppet-lint 4.2.1 and newer).

bastelfreak committed:
Drop broken coverage reporting

We don't use it anymore and it fails in CI.
